pelican/generators.py

```diff
@@ -560,57 +560,69 @@ def generate_tags(self, write):
         tag_template = self.get_template("tag")
         for tag, articles in self.tags.items():
             dates = [article for article in self.dates if article in articles]
-            write(
-                tag.save_as,
-                tag_template,
-                self.context,
-                tag=tag,
-                url=tag.url,
-                articles=articles,
-                dates=dates,
-                template_name="tag",
-                blog=True,
-                page_name=tag.page_name,
-                all_articles=self.articles,
-            )
+            try:
+                write(
+                    tag.save_as,
+                    tag_template,
+                    self.context,
+                    tag=tag,
+                    url=tag.url,
+                    articles=articles,
+                    dates=dates,
+                    template_name="tag",
+                    blog=True,
+                    page_name=tag.page_name,
+                    all_articles=self.articles,
+                )
+            except RuntimeError as e:
+                logger.error('Trying to write Tag page for "%s".' % (tag))
+                raise e
```

Comment:
* It's not "trying" anymore. It tried and now failed, maybe `"Unable to write..."` instead
* Please don't format logs in place (see: [docs](https://docs.getpelican.com/en/latest/contribute.html#logging-tips)): `logger.error("...", tag)`
* You don't do anything with exception (`as e`), so same as:
```python
try:
    # ...
except RuntimeError:
    # ...
    raise
```

Same for the others.